File: backends/python-fastapi/app/domain/gate/value_objects.py
from dataclasses import dataclass
from typing import Any


@dataclass(frozen=True)
class GateIssueVo:
    """Gate JWT 発行バリューオブジェクト。"""
    token: str


@dataclass(frozen=True)
class GateVerifyVo:
    """Gate JWT 検証バリューオブジェクト。"""
    identifier: str
    member: str
    fingerprint: str
    payload: dict

File: backends/ts-hono/src/domain/gate/valueObjects.ts
/**
 * Gate ドメイン バリューオブジェクトモジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */

/** Gate JWT 発行結果 VO。 */
export interface GateIssueVo {
  token: string;
}

/** Gate JWT 検証結果 VO。 */
export interface GateVerifyVo {
  identifier: string;
  member: string | undefined;
  fingerprint: string | null;
  payload: Record<string, unknown>;
}
